### PR TITLE
Feature/lfx 1843 work history

### PR DESCRIPTION
**Pull request by emlimlf**
# Changes proposed ✍️

### What
Implemented a new timeline UI for the contributor's work history based on this design: https://www.figma.com/design/IKlC7mpFdr0GtiSH182KIX/1.-LFX-CM-%7C-Product---Core?node-id=1571-37749&node-type=frame&t=6kxre7hEcRK031S6-0

### Why
To improve the work history section, we will display the organizations and roles similar to what LinkedIn does.

Ticket: [https://linear.app/lfx/issue/LFX-1843/improve-work-history-timeline](https://linear.app/lfx/issue/LFX-1843/improve-work-history-timeline)

## Checklist ✅
- [ ] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screenshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

## Summary by CodeRabbit

- **New Features**
	- Introduced a structured timeline format for displaying contributor work history.
	- Added new components for timeline representation: `LfTimeline` and `LfTimelineItem`.
	- Created a new interface for organizing timeline data.

- **Bug Fixes**
	- Enhanced error handling for displaying messages when no work experiences are available.

- **Style**
	- Implemented new styles for the timeline component, improving visual organization.

- **Documentation**
	- Added Storybook configuration for the `LfTimeline` component to showcase its usage.

### CODE REVIEW

File: frontend/src/ui-kit/timeline/Timeline.stories.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    setup() {
+      return { args };
+    },
+    template: `<lf-timeline :groups="args.groups" v-slot="{ group }">
+      <lf-timeline-item data="Item 1" v-for="item in group.items" :key="item.id">
+        <div class="text-small text-gray-900 mb-1.5 flex items-center gap-1.5">
+          <lf-svg name="id-card" class="h-4 w-4 text-gray-400" />
+          <p class="truncate" style="max-width: 30ch">
+            {{ item.label }}
+          </p>
```
Comment (joanagmaia) [inline frontend/src/ui-kit/timeline/Timeline.stories.ts:42]: This should have by default a text-gray-900 color, and only on hover a blue one (link behaviour)

File: frontend/src/modules/contributor/components/details/work-history/contributor-details-work-history-item.vue [hunk 2]
```diff
@@ -4,36 +4,7 @@
...
-              projectGroup: selectedProjectGroup?.id,
-            },
-          }"
-          class="font-semibold text-medium leading-6 mb-1 truncate text-black hover:text-primary-500 transition block w-full overflow-hidden"
-        >
-          {{ props.organization.displayName }}
-        </router-link>
-
+      <div class="flex flex-auto flex-col overflow-hidden">
         <div v-if="props.organization?.memberOrganizations?.title" class="text-small text-gray-500 mb-1.5 flex items-center gap-1.5">
```
Comment (joanagmaia) [inline frontend/src/modules/contributor/components/details/work-history/contributor-details-work-history-item.vue:8]: This color should be updated to text-gray-900

File: frontend/src/modules/contributor/components/details/contributor-details-work-history.vue [hunk 3]
```diff
@@ -55,7 +57,7 @@
     </div>
 
     <lf-button
-      v-if="!masked && orgs.length > 3"
+      v-if="!masked && orgGrouped.length > minimumShownGroups"
       type="primary-link"
       size="medium"
       class="mt-6"
```
Comment (joanagmaia) [inline frontend/src/modules/contributor/components/details/contributor-details-work-history.vue:63]: This button is not aligned properly with each one of the experiences. It should have space between the experience and the button. I show it in the video

Comment (joanagmaia) [reply frontend/src/modules/contributor/components/details/contributor-details-work-history.vue:63]: Also, there is one behaviour which I'm not sure about the button position. Will ask Nuno

File: frontend/src/ui-kit/timeline/Timeline.stories.ts (hunk 1 above)
Comment (emlimlf) [reply frontend/src/ui-kit/timeline/Timeline.stories.ts:42]: This is just a sample content inside the timeline component. You can theoretically use anything inside the content of the lf-timeline-item. I've added the text-gray-900 anyway.

File: frontend/src/modules/contributor/components/details/contributor-details-work-history.vue [hunk 4]
```diff
@@ -88,19 +90,52 @@ import LfContributorDetailsWorkHistoryItem
...
+      query: {
+        projectGroup: selectedProjectGroup.value?.id,
+      },
+    },
+    icon: grouped[id][0].logo,
+    items: grouped[id],
+  }));
+});
+const minimumShownGroups = computed(() => {
+  const MIN_ITEMS = 3;
```
[the remark this answers is not in the document]
Comment (emlimlf) [reply frontend/src/modules/contributor/components/details/contributor-details-work-history.vue]: I sent a message through slack regarding this. I reverted this anyway to the old fixed 3 organization limit regardless of how many roles the contributor had in that org.

File: frontend/src/modules/contributor/components/details/contributor-details-work-history.vue [hunk 5]
```diff
@@ -31,14 +31,16 @@
...
-      <lf-contributor-details-work-history-item
-        v-for="org of (orgs || []).slice(0, showMore ? (orgs || []).length : 3)"
-        :key="org.id"
-        :contributor="props.contributor"
-        :organization="org"
-        @edit="isEditModalOpen = true; editOrganization = org"
-      />
-      <div v-if="orgs.length === 0" class="pt-2 flex flex-col items-center">
+      <lf-timeline v-slot="{ group }" :groups="shownGroups">
+        <lf-timeline-item v-for="item in group.items" :key="item.id" data="Item 1">
```
[the remark this answers is not in the document]
Comment (emlimlf) [reply frontend/src/modules/contributor/components/details/contributor-details-work-history.vue]: pushed a fix

**Comment by emlimlf** [comment]
@joanagmaia I've addressed the items you highlighted

**Review by joanagmaia — COMMENTED** [review]
All good. Just one last fix:
We shouldn't exceed the width of that section.
Check image 2 - "..." button is vertically aligned with "+" button. 
Check image 1 - since job title is big, the "..." button is overflowing the section, there's also no space between the title and the button

So we should:
- Have a gap space between title and button
- truncate job title to prevent the item from overflowing the section

Screenshot 2024-12-13 at 14 20 13
Screenshot 2024-12-13 at 14 20 18

**Comment by emlimlf** [comment]
@joanagmaia I've pushed an update to fix that last bit you highlighted.

**Review by joanagmaia — APPROVED** [review]

**Pull request merged**